internal/controller/operator/deployment.go

```diff
@@ -1032,4 +1107,42 @@ func buildIdpVolumes(ldapCACert string, routerCertSecret string) []corev1.Volume
 			},
 		},
 	}
+	if len(auditSecretName) > 0 {
```

Comment:
If you change the `AuditSecret` field to be `*string`, you will need to do a `nil` check here.